crates/ruff_cli/src/diagnostics.rs

```diff
@@ -474,14 +472,24 @@ pub(crate) fn lint_stdin(
         );
     }
 
+    let notebook_indexes = if let SourceKind::IpyNotebook(notebook) = source_kind {
+        FxHashMap::from_iter([(
+            path.map_or_else(|| "-".into(), |path| path.to_string_lossy().to_string()),
+            // Index needs to be computed always to store in cache.
+            notebook.index().clone(),
```

Comment:
Nit: Add an `into_index()` method to avoid cloning the notebook

Reply:
Added.